src/components/RegistryAdmin.tsx

```diff
@@ -1,110 +1,163 @@
+  // displayed.
   const logout = () => {
     setAccessToken('');
     Cookies.remove('refreshToken');
+    setIsLoading(false);
   };
 
   const refresh = () => {
     // Look for the refreshToken cookie.
     const refreshToken = Cookies.get('refreshToken');
 
-    // If a refreshToken exists, send it to the refresh endpoint.
+    // If a refreshToken exists, send it as a header to the refresh endpoint.
     if (refreshToken) {
-      fetch(process.env.QA_REFRESH as RequestInfo | URL, {
+      fetch(process.env.REFRESH as RequestInfo | URL, {
         method: 'POST',
         headers: { Authorization: `Bearer ${refreshToken}` },
       })
         .then((response) => {
           if (response.ok) {
             return response.json();
           } else {
-            // If the server responds with an error, setIsLoading to false.
-            // This will render the LoginForm.
-            setIsLoading(false);
             throw new Error('There was a problem refreshing.');
           }
         })
-        // If the server responds with a new accessToken, save it to
-        // tokenContext. Then, setIsLoading to false. This will render the
-        // LibrariesList.
+        // If the call is successful, save the new accessToken to tokenContext.
         .then((data) => {
           setAccessToken(data.access_token);
           setIsLoading(false);
         })
-        .catch((err) => console.log(err.message));
+        // If there is a problem refreshing, log the error message and use the
+        // logout function to delete the accessToken (if one exists) and the
+        // refreshToken. This will bring the user back to the LoginForm.
+        .catch((err) => {
+          console.log(err.message);
+          logout();
+        });
     } else {
-      // If there is no refreshToken, setIsLoading to false.
-      // This will render the LoginForm.
-      setIsLoading(false);
+      // If no refreshToken exists, delete the accessToken (if there is one)
+      // so the user is brought back to the LoginForm.
+      logout();
     }
   };
 
-  // If there is no accessToken in tokenContext, try refreshing. Else,
-  // setIsLoading to false. This will render the LibrariesList.
+  // Fetch the libraries with the accessToken.
+  const fetchLibraries = () => {
+    fetch(process.env.FETCH_LIBRARIES as RequestInfo | URL, {
+      headers: { Authorization: `Bearer ${accessToken}` },
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+          // If the response status is 401, the accessToken may be expired.
+          // Try refreshing it.
+        } else if (response.status === 401) {
+          refresh();
+          return;
+        } else {
+          throw new Error('There was a problem fetching libraries.');
+        }
+      })
+      .then((data) => {
+        setLibraries(data.libraries);
+        setIsLoading(false);
+      })
+      .catch((err) => console.log(err.message));
```

Comment:
Should we `setIsLoading(false)` here too and render an error component? This would be for a future ticket & PR. But want to get your thoughts on what you think might be a good way to display errors.

Reply:
Yeah, good callout. I have a ticket in the backlog regarding error handling.